presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -273,7 +273,7 @@ PlanNodeStatsEstimate calculateJoinComplementStats(
             PlanNodeStatsEstimate rightStats,
             TypeProvider types)
     {
-        if (rightStats.getOutputRowCount() == 0) {
+        if (leftStats.getOutputRowCount() == 0 || rightStats.getOutputRowCount() == 0) {
```

Comment:
Can we have a test for that? Especially in one of the `TestTpc*Stats` classes?

Reply:
Since we are not going to round, explicit check `== 0` is misleading. Given that there are no divisions or multiplications by the rows count it looks safe to just remove this special case check altogether.

Reply:
I'm reverting this change. As there's no division or multiplication involving `leftStats.getOutputRowCount()`